flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/sink/writer/topic/TopicProducerRegister.java

```diff
@@ -150,7 +181,13 @@ private <T> Producer<T> getOrCreateProducer(String topic, Schema<T> schema) {
                     createProducerBuilder(pulsarClient, schema, sinkConfiguration);
             // Set the required topic name.
             builder.topic(topic);
+            // Set the sending counter for metrics.
+            builder.intercept(new ProducerMetricsInterceptor(metricGroup));
```

Comment:
Do we also exclude this interceptor if metrics are disabled?

Reply:
This interceptor is only used for recording the IO metrics and no need to disable it.

Reply:
OK. It's fine if no significant performance regression.

Reply:
Although I think logically we should skip adding this interceptor if metrics are disabled.